k-distribution/k-tutorial/1_basic/03_parsing/README.md

```diff
@@ -1 +1,627 @@
+this, the program we mentioned above successfully parses.
+
+## Associativity
+
+Even having broken the expression grammar into priority blocks, the resulting
+grammar is still ambiguous. We can see this if we try to parse the following
+program (`assoc.bool`):
+
+```
+true && false && false
+```
+
+Priority blocks will not help us here: the problem comes between two parses
+where both possible parses have a direct parent and child which is within a
+single priority block.
+
+This is where the notion of associativity comes into play. Essentially,
+associativity applies the following additional rule to parses: a
+left-associative symbol cannot appear as a direct rightmost child of a symbol
+with equal priority; a right-associative symbol cannot appear as a direct
+leftmost child of a symbol with equal priority; and a non-associative symbol
+cannot appear as a direct leftmost **or** rightmost child of a symbol with
+equal priority.
+
+In C, binary operators are all left-associative, meaning that the expression 
+`true && false && false` parses unambiguously as `(true && false) && false`,
+because `&&` cannot appear as the rightmost child of itself.
+
+Consider, then, the fourth iteration on the grammar of this definition
+(`lesson-03-f.k`):
+
+```k
+module LESSON-03-F
+
+  syntax Boolean ::= "true" | "false"
+                   | "(" Boolean ")" [bracket]
+                   > "!" Boolean [function]
+                   > left: Boolean "&&" Boolean [function]
+                   > left: Boolean "^" Boolean [function]
+                   > left: Boolean "||" Boolean [function]
+
+endmodule
+```
+
+Here each priority group, immediately after the `::=` or `>` operator, can
+be followed by a symbol representing the associativity of that priority group:
+either `left:` for left associativity, `right:` for right associativity, or
+`non-assoc:` for non-associativity. In this example, each priority group we
+apply associativity to has only a single production, but we could equally well
+write a priority block with multiple productions and an associativity.
+
+For example, consider the following, different grammar (`lesson-03-g.k`):
+
+```k
+module LESSON-03-G
+
+  syntax Boolean ::= "true" | "false"
+                   | "(" Boolean ")" [bracket]
+                   > "!" Boolean [function]
+                   > left: 
+                     Boolean "&&" Boolean [function]
+                   | Boolean "^" Boolean [function]
+                   | Boolean "||" Boolean [function]
+
+endmodule
+```
+
+In this example, unlike the one above, `&&`, `^`, and `||` have the same
+priority. However, viewed as a group, the entire group is left associative.
+This means that none of `&&`, `^`, and `||` can appear as the right child of
+any of `&&`, `^`, or `||`. As a result of this, this grammar is also not
+ambiguous. However, it expresses a different grammar, and you are encouraged
+to think about what the differences are in practice.
```

Comment:
Please provide some example programs here that users can call `kast` on and see the difference, rather than just saying "think about it".